File: backend/src/main/java/eu/fuelfleet/customer/service/CustomerService.java
package eu.fuelfleet.customer.service;

import eu.fuelfleet.auth.entity.User;
import eu.fuelfleet.auth.repository.UserRepository;
import eu.fuelfleet.customer.dto.CustomerRequest;
import eu.fuelfleet.customer.dto.CustomerResponse;
import eu.fuelfleet.customer.entity.Customer;
import eu.fuelfleet.customer.repository.CustomerRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;
import java.util.List;
import java.util.UUID;

@Service
@RequiredArgsConstructor
public class CustomerService {
    private final CustomerRepository customerRepository;
    private final UserRepository userRepository;

    public List<CustomerResponse> getAll() {
        return customerRepository.findByCompanyId(getCompanyId()).stream()
                .map(CustomerResponse::from).toList();
    }

    public List<CustomerResponse> getActive() {
        return customerRepository.findByCompanyIdAndActiveTrue(getCompanyId()).stream()
                .map(CustomerResponse::from).toList();
    }

    @Transactional
    public CustomerResponse create(CustomerRequest req) {
        User user = getCurrentUser();
        Customer c = Customer.builder()
                .company(user.getCompany()).name(req.getName()).vatNumber(req.getVatNumber())
                .regCode(req.getRegCode()).email(req.getEmail()).phone(req.getPhone())
                .address(req.getAddress()).country(req.getCountry())
                .contactPerson(req.getContactPerson())
                .paymentTermDays(req.getPaymentTermDays()).notes(req.getNotes()).active(true)
                .build();
        return CustomerResponse.from(customerRepository.save(c));
    }

    @Transactional
    public CustomerResponse update(UUID id, CustomerRequest req) {
        Customer c = customerRepository.findById(id)
                .filter(cu -> cu.getCompany().getId().equals(getCompanyId()))
                .orElseThrow(() -> new RuntimeException("Customer not found"));
        c.setName(req.getName()); c.setVatNumber(req.getVatNumber());
        c.setRegCode(req.getRegCode()); c.setEmail(req.getEmail());
        c.setPhone(req.getPhone()); c.setAddress(req.getAddress());
        c.setCountry(req.getCountry()); c.setContactPerson(req.getContactPerson());
        c.setPaymentTermDays(req.getPaymentTermDays()); c.setNotes(req.getNotes());
        return CustomerResponse.from(customerRepository.save(c));
    }

    @Transactional
    public void delete(UUID id) {
        customerRepository.findById(id)
                .filter(c -> c.getCompany().getId().equals(getCompanyId()))
                .ifPresent(customerRepository::delete);
    }

    private UUID getCompanyId() { return getCurrentUser().getCompany().getId(); }
    private User getCurrentUser() {
        String email = SecurityContextHolder.getContext().getAuthentication().getName();
        return userRepository.findByEmail(email).orElseThrow();
    }
}

File: frontend/src/app/shared/services/customer.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface Customer {
  id?: string;
  name: string;
  vatNumber?: string;
  regCode?: string;
  email?: string;
  phone?: string;
  address?: string;
  country?: string;
  contactPerson?: string;
  paymentTermDays?: number;
  active?: boolean;
  notes?: string;
}

@Injectable({ providedIn: 'root' })
export class CustomerService {
  private http = inject(HttpClient);
  private api = '/api/customers';

  getAll(): Observable<Customer[]> { return this.http.get<Customer[]>(this.api); }
  getActive(): Observable<Customer[]> { return this.http.get<Customer[]>(`${this.api}/active`); }
  create(customer: Customer): Observable<Customer> { return this.http.post<Customer>(this.api, customer); }
  update(id: string, customer: Customer): Observable<Customer> { return this.http.put<Customer>(`${this.api}/${id}`, customer); }
  delete(id: string): Observable<void> { return this.http.delete<void>(`${this.api}/${id}`); }
}
